catroid/src/main/java/org/catrobat/catroid/content/Project.java

```diff
@@ -277,7 +277,7 @@ public void setCatrobatLanguageVersion(float catrobatLanguageVersion) {
 	public void setDeviceData(Context context) {
 		// TODO add other header values
 		xmlHeader.setPlatform(Constants.PLATFORM_NAME);
-		xmlHeader.setPlatformVersion((double) Build.VERSION.SDK_INT);
+		xmlHeader.setPlatformVersion(String.valueOf(Build.VERSION.SDK_INT));
```

Comment:
Since the platform version changed from double to String, Android version 26 will be stored as `<platformVersion>26</platformVersion>`, not as previously `<platformVersion>26.0</platformVersion>` (trailing zero!)

Reply:
Will that make any problems with old projects, or new ones?

Reply:
> Will that make any problems with old projects, or new ones?

No, old Pocket Code apps will just store `26.0` whereas new ones will store `26`.
All Pocket Code apps are able to parse `26.0` and `26` since both can be parsed as Double or String.
___
More detailed explanation: The error message mentioned in CATROID-65
```
org.catrobat.catroid E/BackwardCompatibleCatrobatLanguageXStream: Conversion error For input string: "25a" : For input string: "25a"
    ---- Debugging information ----
    message             : For input string: "25a"
    cause-exception     : java.lang.NumberFormatException
    cause-message       : For input string: "25a"
    class               : java.lang.Double
    required-type       : java.lang.Double
    converter-type      : com.thoughtworks.xstream.converters.SingleValueConverterWrapper
    wrapped-converter   : com.thoughtworks.xstream.converters.basic.DoubleConverter
    path                : /program/header/platformVersion
    line number         : 16
    class[1]            : org.catrobat.catroid.content.XmlHeader
    converter-type[1]   : com.thoughtworks.xstream.converters.reflection.ReflectionConverter
    class[2]            : org.catrobat.catroid.content.Project
    version             : 0.0
    -------------------------------
```

So XStream uses [com.thoughtworks.xstream.converters.basic.DoubleConverter.java#L26-L29](https://github.com/x-stream/xstream/blob/a925ad0642f7bd6e188e27d61c6e7123dff119ad/xstream/src/java/com/thoughtworks/xstream/converters/basic/DoubleConverter.java#L26-L29) to parse the `platformVersion` as double
```java
@Override
public Object fromString(final String str) {
    return Double.valueOf(str);
}
```

When changing the `platformVersion` type from double to String, the `DoubleConverter` isn't used anymore, of course, and the parsed String is directly stored.

Reply:
Just to Highlight: Once the `platformVersion` content isn't a parseable double anymore, e.g. Catty changes the format to MAJOR.MINOR.PATCH, old Pocket Code apps won't be able to open such programs.

When Catty should use a non numeric format like MAJOR.MINOR.PATCH all depends on how many and quickly users update Catroid to the latest version but also how many Catty programs are uploaded to the share. Since I don't have this data, I can't say when Catty should change their `platformVersion` format. However, the sooner this pull request is released, the sooner Catty can adapt their `platformVersion` to a non numeric value.

Summarized:
* Old and new Catroid apps have no problems loading programs created by an old or new Catroid app.
* Old Catroid apps will have problems once the `platformVersion` isn't a parseable double anymore.